simulator-core/src/test/java/com/evbox/everon/ocpp/simulator/mock/RequestExpectationMatcher.java

```diff
@@ -0,0 +1,61 @@
+package com.evbox.everon.ocpp.simulator.mock;
+
+import com.evbox.everon.ocpp.simulator.message.Call;
+import lombok.Data;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Represents an expectation of the request.
+ */
+@Data
+public class RequestExpectationMatcher {
+
+    private final Predicate<Call> expectedRequest;
+    private final ExpectedCount expectedCount;
+
+    private final Function<Call, String> expectedResponse;
+
+    private final AtomicInteger actualCount = new AtomicInteger(0);
```

Comment:
nit: i don't think you really need an atomic counter here. regular one works fine.

i mean what race condition are you guarding against in your code, if this was just a regular counter?

Reply:
yeah i understand that, but what i'm saying is that if you replaced this atomic counter with a non atomic one, it would be fine (you wouldn't get into problem).

That said it's a small issue, so it doesn't matter, we can leave it as is.

Reply:
So what is your proposal? volatile, synchronized? Just curious...